src/longrange_force_types/force_ewald_cabana_neigh_impl.h

```diff
@@ -0,0 +1,597 @@
+                                  U_trigonometric( 2 * kidx ) +
+                              U_trigonometric( 2 * kidx + 1 ) *
+                                  U_trigonometric( 2 * kidx + 1 ) );
+                        p( idx ) += contrib;
+                        //Uk_part += contrib;
+
+                        for ( int dim = 0; dim < 3; ++dim )
+                            f( idx, dim ) +=
+                                k_coeff * 2.0 * q( idx ) * k[dim] *
+                                ( U_trigonometric( 2 * kidx + 1 ) * cos( kr ) -
+                                  U_trigonometric( 2 * kidx ) * sin( kr ) );
+                }
+            }
+        }
+    };
+    Kokkos::parallel_for( Kokkos::RangePolicy<ExecutionSpace>( 0, N_max ), kspace_potential );
+    Kokkos::fence();
+
+    //MPI_Allreduce( MPI_IN_PLACE, &Uk, 1, MPI_DOUBLE, MPI_SUM, comm );
+
+    // computation real-space contribution
+    //
+    // In order to compute the real-space contribution to potentials and
+    // forces the Cabana implementation of halos and Verlet lists is
+    // used. The halos are used to communicate particles along the
+    // borders of MPI domains to their respective neighbors, so that
+    // complete Verlet lists can be created. To save computation time
+    // the half shell variant is used, that means that Newton's third
+    // law of motion is used: F(i,j) = -F(j,i). The downside of this
+    // is that the computed partial forces and potentials of the
+    // ghost particles need to be communicated back to the source
+    // process, which is done by using the 'scatter' implementation
+    // of Cabana.
+    // TODO: Enable re-use of CabanaMD neighborlist from short-range forces
+
+    
+    // lower end of system
+    double grid_min[3] = {system->sub_domain_lo_x - system->sub_domain_x,
+                          system->sub_domain_lo_y - system->sub_domain_y,
+                          system->sub_domain_lo_z - system->sub_domain_z};
+    // upper end of system
+    double grid_max[3] = {system->sub_domain_hi_x + system->sub_domain_x,
+                          system->sub_domain_hi_y + system->sub_domain_y,
+                          system->sub_domain_hi_z + system->sub_domain_z};
+
+    //double grid_min[3] = {-r_max + domain_length( 0 ),
+    //                      -r_max + domain_length( 2 ),
+    //                      -r_max + domain_length( 4 )};
+
+    using ListAlgorithm = Cabana::HalfNeighborTag;
+    using ListType =
+        Cabana::VerletList<DeviceType, ListAlgorithm, Cabana::VerletLayoutCSR>;
+
+    // store the number of local particles
+    //int n_local = n_max;
+    // offset due to the previously received particles
+    int offset = 0;
+
+    // communicate particles along the edges of the system
+
+    // six halo regions required for transport in all directions
+    std::vector<int> n_export = {0, 0, 0, 0, 0, 0};
+    std::vector<Cabana::Halo<DeviceType> *> halos( 6 );
```

Comment:
I think this should be done with Comm. This has to be done for short range forces, so if long range is just incremented to the short range, you shouldn't need to do anything per atom. We do have to be careful about where/when we reset forces/energies to zero (or split into short/long)